tests/processes/test_wps_package.py

```diff
@@ -494,3 +500,155 @@ def test_merge_io_formats_wps_overlaps_cwl():
         Format(CONTENT_TYPE_APP_XML),
         Format(CONTENT_TYPE_TEXT_PLAIN),
     ])
+
+
+def test_stdout_stderr_logging_for_commandline_tool_success():
+    process = Process({
+        "title": "test-stdout-stderr",
+        "id": "test-stdout-stderr",
+        "package": {
+            "cwlVersion": "v1.0",
+            "class": "CommandLineTool",
+            "baseCommand": "echo",
+            "inputs": {
+                "message": {
+                    "type": "string",
+                    "inputBinding": {
+                        "position": 1
+                    }
+                }
+            },
+            "outputs": {
+
+            }
+        }
+    })
+
+    payload = process
+    package = process["package"]
+    title = process["title"]
+    identifier = process["id"]
+    wps_package_instance = WpsPackage(identifier=identifier, title=title, payload=payload, package=package)
+
+    # WPSRequest mock
+    wps_request = WPSRequest()
+    wps_request.json = {
+        "identifier": "test-stdout-stderr",
+        "operation": "execute",
+        "version": "1.0.0",
+        "language": "null",
+        "identifiers": "null",
+        "store_execute": "true",
+        "status": "true",
+        "lineage": "true",
+        "raw": "false",
+        "inputs": {
+            "message":[
+                 {
+                    "identifier":"message",
+                    "title":"A dummy message",
+                    "type":"literal",
+                    "data_type":"string",
+                    "data":"Dummy message",
+                    "allowed_values": [
+
+                    ],
+                 }
+            ]
+        },
+        "outputs":{
+
+        }
+    }
+
+    # ExecuteResponse mock
+    wps_response = type('',(object,),{"_update_status": lambda w,x,y,z: 1 })()
+
+    # WPSPackage._handle()
+    log_file = tempfile.NamedTemporaryFile()
+    status_location = log_file.name
+    workdir = tempfile.TemporaryDirectory()
+    wps_package_instance.status_location = status_location          # to retrieve logs
+    wps_package_instance.workdir = workdir.name
+    wps_package_instance._handler(wps_request, wps_response)
+
+    # log assertions
+    with open(status_location + ".log", "r") as file:
+        log_data = file.read()
+        assert "Dummy message" in log_data
+
+
+def test_stdout_stderr_logging_for_commandline_tool_failure():
```

Comment:
criteria docstring

Reply:
Also, we could make a common function that setups all the boilerplate code, and only replace the `not_existing_command` vs `echo` in each case to remove duplication.

Reply:
Good idea, this amelioration would be done at the same time we add the stdout/stderr for the workflows